WM4 - Kerim Zamir - JS Core 1 - Week 3

Arbeits-Sachen opened the pull request.

bradleymubenga777 commented:
Hey Khaya! Things Look Great However You Still Need To Fill In The Homework Section Buddy.

Arbeits-Sachen committed:
Update exercise.js

Arbeits-Sachen committed:
Update exercise.js

Arbeits-Sachen committed:
Update exercise.js

Arbeits-Sachen committed:
Update exercise.js

*note* for doesnt work

Arbeits-Sachen committed:
Update exercise.js

Arbeits-Sachen committed:
Update exercise1.js

Arbeits-Sachen committed:
Update exercise2.js

Arbeits-Sachen committed:
Update exercise.js

*note* const is not needed before i in for of loop

Arbeits-Sachen committed:
Update 1-weather-report.js

Arbeits-Sachen committed:
Update 2-retrying-random-numbers.js

Arbeits-Sachen committed:
Update 3-financial-times.js

Arbeits-Sachen committed:
Update 4-stocks.js

*note* const average = (closingPricesForAllStocks.map((array) => array.reduce((r, current) => r + current / array.length, 0))).map((element) => parseFloat(element.toFixed(2)));
Should be used more often

Arbeits-Sachen committed:
Update 1-factorial.js

SaadiaELF commented on 1-exercises/B-while-loop/exercise.js:
```diff
@@ -5,8 +5,24 @@
 
-function evenNumbers(n) {
-    // TODO
+function evenNumbers(n)
+{
+    let total = "0";
+    let calc = 0;
+    
+    if (n === 0)
+    {
+        return console.log("");
+    }
+
+    for(let i = 1; i < n; i++)
+    {
+        calc = calc + 2;
+
+        total = total + ", " + calc;
+    }
+
+    return console.log(total);
```
The instructions said that you should use a while loop.
Consider using an array to store the values and then convert it to string using 'toString' method

SaadiaELF commented on 1-exercises/E-for-loop/exercise2.js:
```diff
@@ -27,7 +27,10 @@ const AGES = [
-
+for (let i = 0; i < WRITERS.length; i++)
+{
+    console.log(WRITERS[i] + " is " + AGES[i] + " old")
```
```suggestion
    console.log(WRITERS[i] + " is " + AGES[i] + "years old")
```

SaadiaELF commented on 1-exercises/F-for-of-loop/exercise.js:
```diff
@@ -11,6 +11,15 @@ let tubeStations = [
     "Tottenham Court Road"
 ];
 
+for (i of tubeStations)
```
```suggestion
for (const station of tubeStations)
```
I think it will be better to use a meaningful variables names, it's easy to read and understand ("i" generally refers to index which is not the case here)

SaadiaELF commented on 2-mandatory/1-weather-report.js:
```diff
@@ -11,8 +11,15 @@
 
-function getTemperatureReport(cities) {
-    // TODO
+function getTemperatureReport(cities)
```
Good work!

SaadiaELF commented on 2-mandatory/2-retrying-random-numbers.js:
```diff
@@ -9,8 +9,16 @@ function generateRandomNumber() {
 
-function getRandomNumberGreaterThan50() {
-    // TODO - implement using a do-while loop
+function getRandomNumberGreaterThan50()
+{
+    let total;
+    do
+    {
+        (total = generateRandomNumber());
+    }
+    while (total <= 50);
+
+    return total;
```
I think the use of 'total' as the name of the variable creates a little confusion as we don't really calculate any total in this function

SaadiaELF commented on 2-mandatory/3-financial-times.js:
```diff
@@ -1,40 +1,72 @@
-function titleWithFewestWords(allArticleTitles) {
-    // TODO
+
+function titleWithFewestWords(allArticleTitles)
+{
+    const shorter = (left, right) => left.length <= right.length ? left : right;
+
+    return allArticleTitles.reduce(shorter);
 }
 
```
Could you please add some comments to this function, I didn't really understand it?

SaadiaELF commented on 2-mandatory/4-stocks.js:
```diff
@@ -33,8 +33,14 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
 */
-function getAveragePrices(closingPricesForAllStocks) {
-    // TODO
+//https://jrsinclair.com/articles/2019/five-ways-to-average-with-js-reduce/
+function getAveragePrices(closingPricesForAllStocks)
+{
+    //const reduce  = r => i => a => a.reduce(r, i);
+    //conatiner   =     getting items                      => got items.reduce          =>      add    /    length         .map previous  =>       round to 2 dec
+    const average = (closingPricesForAllStocks.map((array) => array.reduce((r, current) => r + current / array.length, 0))).map((element) => parseFloat(element.toFixed(2)));
+
+    return average;
```
Could you please delete unnecessary comments to make the code cleaner :D

SaadiaELF commented on 2-mandatory/4-stocks.js:
```diff
@@ -47,8 +53,12 @@ function getAveragePrices(closingPricesForAllStocks) {
-function getPriceChanges(closingPricesForAllStocks) {
-    // TODO
+
+function getPriceChanges(closingPricesForAllStocks)
+{
+    const PriceChange = (closingPricesForAllStocks.map((array) => array[array.length - 1] - array[0])).map((element) => parseFloat(element.toFixed(2)));
+
+    return PriceChange;
```
```suggestion
function getPriceChanges(closingPricesForAllStocks)
{
    const priceChange = closingPricesForAllStocks.map((array) => array[array.length - 1] - array[0])
    const result = priceChange.map((element) => parseFloat(element.toFixed(2));
    return result;
```
It's better to use to variables to make code more readable

SaadiaELF reviewed (commented):
Good work 👏 
I have just remarked that you are using a capital first letter in variables name, I'm not sure if it's the right naming convention for JS. And also variable name should be meaningful

Alex-Phillip commented on 1-exercises/B-while-loop/exercise.js:
```diff
@@ -5,8 +5,24 @@
+        return console.log("");
+    }
+
+    for(let i = 1; i < n; i++)
```
Hi Kerim, I've looked through your code and it's great! One suggestion I have here however is using a while loop instead of a for loop.
Great work!
